src/mult.jl

```diff
@@ -60,7 +71,7 @@ end
 function _term_poly_mult(t::Term{C, S}, p::Polynomial{C, T}, op::Function) where {C, S, T}
     U = MA.promote_operation(op, S, T)
     if iszero(t)
-        zero(Polynomial{C, U})
+        zero(p)
```

Comment:
Use `zero_with_variables(Polynomial{C, U}, variables(p))`

Reply:
My first guess was wrong, this was not the culprit. I don't know whether to keep `zero(Polynomial{C, U})` or use `zero_with_variables( Polynomial{C,U}, variables(p) )`.

Reply:
You can leave as it is then. We can make the change in another PR.